clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -157,9 +162,28 @@ protected CompletableFuture<V> get(GetRequestContext requestContext, K key) thro
     }
 
     /**
-     * This atomic variable is used to indicate whether a faster response has returned or not.
+     * This atomic variable is used to find the fastest response received and ignore other responses
      */
     AtomicBoolean receivedSuccessfulResponse = new AtomicBoolean(false);
+
+    /**
+     * List of futures used below and their relationships:
+     * 1. valueFuture => this is the completable future that is returned from this function which either throws an exception or returns null or returns value.
+     * 2. routeRequestFuture => an incomplete completable future returned from {@link com.linkedin.venice.fastclient.meta.InstanceHealthMonitor}
+     *                          for {@link com.linkedin.venice.fastclient.meta.AbstractStoreMetadata} by adjusting
+     *                          {@link com.linkedin.venice.fastclient.meta.InstanceHealthMonitor#pendingRequestCounterMap} for each server instances per
+     *                          store before starting a get request and during completing this future.
+     *                          This is also added to requestContext.routeRequestMap to indicate whether a particular server instance is already
+     *                          used for this get request: to not reuse the same instance for both the queries for a key when speculative query
+     *                          is enabled. TODO what if there are only 2 replicas and both the replicas are in the same server?
```

Comment:
```
TODO what if there are only 2 replicas and both the replicas are in the same server?
```
This is not possible.

Reply:
Is this because Helix takes care of this?